src/main/java/seedu/address/storage/JsonAdaptedParticipation.java

```diff
@@ -72,14 +80,17 @@ public Participation toModelType(AddressBook addressBook) throws IllegalValueExc
         if (student == null) {
             throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Person.class.getSimpleName()));
         }
+        if (phone == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Person.class.getSimpleName()));
+        }
         if (tutorial == null) {
             throw new IllegalValueException(
                     String.format(MISSING_FIELD_MESSAGE_FORMAT, Tutorial.class.getSimpleName()));
         }
 
         // Matches parsed json strings to objects in addressbook
         Person studentObject = addressBook.getPersonList()
-                .filtered(p -> p.getName().fullName.equals(student))
+                .filtered(p -> p.getName().fullName.equals(student) && p.getPhoneValue().equals(phone))
```

Comment:
I think we should use equalsIgnoreCase when checking the name of the person. okay but maybe this should be in the equals method of name
<img width="995" alt="Screenshot 2024-10-30 at 6 15 22 PM" src="https://github.com/user-attachments/assets/571e256b-8885-4775-846a-ed40fc43e6eb">